htdocs/mrp/ajax/interface.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ *	\file       htdocs/mrp/ajax/interface.php
+ */
+
+if (!defined('NOREQUIRESOC')) {
+	define('NOREQUIRESOC', '1');
+}
+if (!defined('NOCSRFCHECK')) {
+	define('NOCSRFCHECK', '1');
+}
+if (!defined('NOTOKENRENEWAL')) {
+	define('NOTOKENRENEWAL', '1');
+}
+if (!defined('NOREQUIREMENU')) {
+	define('NOREQUIREMENU', '1');
+}
+if (!defined('NOREQUIREHTML')) {
+	define('NOREQUIREHTML', '1');
+}
+if (!defined('NOREQUIREAJAX')) {
+	define('NOREQUIREAJAX', '1');
+}
+
+global $db, $langs;
+
+require '../../main.inc.php'; // Load $user and permissions
+
+$warehouse_id = GETPOST('warehouse_id', 'int');
+$batch_id = GETPOST('batch_id', 'int');
+$fk_product = GETPOST('product_id', 'int');
+$action = GETPOST('action', 'alphanohtml');
+$permissiontoproduce = GETPOST('permissiontoproduce', 'int');
+
+if ($action == 'updateselectbatchbywarehouse' && $permissiontoproduce) {
+	$TRes = array();
+
+	$sql = "SELECT pb.batch, pb.rowid, ps.fk_entrepot, pb.qty, e.ref as label, ps.fk_product";
+	$sql .= " FROM " . MAIN_DB_PREFIX . "product_batch as pb";
+	$sql .= " LEFT JOIN " . MAIN_DB_PREFIX . "product_stock as ps on ps.rowid = pb.fk_product_stock";
+	$sql .= " LEFT JOIN " . MAIN_DB_PREFIX . "entrepot as e on e.rowid = ps.fk_entrepot AND e.entity IN (" . getEntity('stock') . ")";
+	$sql .= " WHERE ps.fk_product = '" . $fk_product . "'";
```

Comment:
Must be 
$sql .= " WHERE ps.fk_product = " . ((int) $fk_product);
Not quote on a comparison with a field that is integer (pb with not optimized sql execution plan due to implicit cast done by database engine).